src/ray/ray_syncer/ray_syncer_client.cc

```diff
@@ -28,9 +28,19 @@ RayClientBidiReactor::RayClientBidiReactor(
     instrumented_io_context &io_context,
     std::function<void(std::shared_ptr<const RaySyncMessage>)> message_processor,
     std::function<void(RaySyncerBidiReactor *, bool)> cleanup_cb,
-    std::unique_ptr<ray::rpc::syncer::RaySyncer::Stub> stub)
+    std::unique_ptr<ray::rpc::syncer::RaySyncer::Stub> stub,
+    bool batching_enabled)
     : RaySyncerBidiReactorBase<ClientBidiReactor>(
-          io_context, remote_node_id, std::move(message_processor)),
+          io_context,
+          remote_node_id,
+          std::move(message_processor),
+          /* batch_size */
+          batching_enabled ? RayConfig::instance().gcs_resource_broadcast_max_batch_size()
+                           : 1,
+          /* batch_delay_ms */
+          batching_enabled
+              ? RayConfig::instance().gcs_resource_broadcast_max_batch_delay_ms()
+              : 0),
```

Comment:
This class is also constructed in the node manager, in which we never want to do the batching. So these parameters should be passed down from `gcs_server.cc`.

Reply:
Note that as written it is also an abstraction leak -- `gcs_blah` should not exist in a shared component.